File: Code 3 Callouts/Code 3 Callouts/Util/Audio/AudioPlayerEngine.vb
Imports Rage
Imports System.IO
Imports System.Threading.Tasks

Namespace Util.Audio

    Module AudioPlayerEngine

        Friend Sub PlayDispatchAudio(ByVal pEnum As [Enum])
            Dim a As New AudioFile("DISPATCH", pEnum)
            PlayAudio(a)
        End Sub

        Friend Sub PlayOfficerAudio(ByVal pEnum As [Enum])
            Dim a As New AudioFile("OFFICER", pEnum)
            PlayAudio(a)
        End Sub

        Friend Sub PlayAudio(ByVal pAudioFile As AudioFile)
            Dim list As New List(Of AudioFile)
            list.Add(pAudioFile)
            PlayAudio(list)
        End Sub

        Friend Sub PlayAudio(ByVal pAudioFiles As List(Of AudioFile), Optional ByVal pLocation As Vector3 = Nothing)
            Try
                GameFiber.StartNew(
                    Sub()
                        Dim mAudioNames As String() = pAudioFiles.Select(Function(x) x.FileEnum.ToString()).ToArray()
                        Dim audio As String = String.Join(" ", mAudioNames)
                        Logger.LogVerboseDebug("Audio String -- " & audio)

                        If pLocation = Nothing Then
                            LSPD_First_Response.Mod.API.Functions.PlayScannerAudio(audio)
                        Else
                            LSPD_First_Response.Mod.API.Functions.PlayScannerAudioUsingPosition(audio, pLocation)
                        End If
                    End Sub)
            Catch ex As Exception
                Logger.LogVerbose("Error playing audio -- " & ex.Message)
            End Try
        End Sub

        Friend Sub PlayAudio(ByVal pAudioFiles As List(Of String), Optional ByVal pLocation As Vector3 = Nothing)
            Try
                GameFiber.StartNew(
                    Sub()
                        Dim audio As String = String.Join(" ", pAudioFiles)

                        If pLocation = Nothing Then
                            LSPD_First_Response.Mod.API.Functions.PlayScannerAudio(audio)
                        Else
                            LSPD_First_Response.Mod.API.Functions.PlayScannerAudioUsingPosition(audio, pLocation)
                        End If
                    End Sub)
            Catch ex As Exception
                Logger.LogVerbose("Error playing audio -- " & ex.Message)
            End Try
        End Sub

        Friend Sub PlayAudio(ByVal pAudio As String, Optional ByVal pLocation As Vector3 = Nothing)
            Try
                GameFiber.StartNew(
                    Sub()
                        If pLocation = Nothing Then
                            LSPD_First_Response.Mod.API.Functions.PlayScannerAudio(pAudio)
                        Else
                            LSPD_First_Response.Mod.API.Functions.PlayScannerAudioUsingPosition(pAudio, pLocation)
                        End If
                    End Sub)
            Catch ex As Exception
                Logger.LogVerbose("Error playing audio -- " & ex.Message)
            End Try
        End Sub

        Friend Class AudioFile
            Public Sub New(ByVal pClassName As String, ByVal pAudioFile As [Enum])
                _class = pClassName
                _fileEnum = pAudioFile
            End Sub

            Private _fileEnum As [Enum]
            Public Property FileEnum() As [Enum]
                Get
                    Return _fileEnum
                End Get
                Set(ByVal value As [Enum])
                    _fileEnum = value
                End Set
            End Property

            Private _class As String
            Public Property AudioClass() As String
                Get
                    Return _class
                End Get
                Set(ByVal value As String)
                    _class = value
                End Set
            End Property
        End Class

    End Module

End Namespace

File: Code 3 Callouts/Code 3 Callouts/Util/Audio/AudioPlayerEngine.cs
using Rage;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace Stealth.Plugins.Code3Callouts.Util.Audio
{

	internal static class AudioPlayerEngine
	{

		internal static void PlayDispatchAudio(Enum pEnum)
		{
			AudioFile a = new AudioFile("DISPATCH", pEnum);
			PlayAudio(a);
		}

		internal static void PlayOfficerAudio(Enum pEnum)
		{
			AudioFile a = new AudioFile("OFFICER", pEnum);
			PlayAudio(a);
		}

		internal static void PlayAudio(AudioFile pAudioFile)
		{
			List<AudioFile> list = new List<AudioFile>();
			list.Add(pAudioFile);
			PlayAudio(list);
		}

		internal static void PlayAudio(List<AudioFile> pAudioFiles, Vector3? pLocation = null)
		{
			try {
				GameFiber.StartNew(() =>
				{
					string[] mAudioNames = pAudioFiles.Select(x => x.FileEnum.ToString()).ToArray();
					string audio = string.Join(" ", mAudioNames);
					Logger.LogVerboseDebug("Audio String -- " + audio);

					if (pLocation == null) {
						LSPD_First_Response.Mod.API.Functions.PlayScannerAudio(audio);
					} else {
						LSPD_First_Response.Mod.API.Functions.PlayScannerAudioUsingPosition(audio, pLocation.Value);
					}
				});
			} catch (Exception ex) {
				Logger.LogVerbose("Error playing audio -- " + ex.Message);
			}
		}

		internal static void PlayAudio(List<string> pAudioFiles, Vector3? pLocation = null)
		{
			try {
				GameFiber.StartNew(() =>
				{
					string audio = string.Join(" ", pAudioFiles);

					if (pLocation == null) {
						LSPD_First_Response.Mod.API.Functions.PlayScannerAudio(audio);
					} else {
						LSPD_First_Response.Mod.API.Functions.PlayScannerAudioUsingPosition(audio, pLocation.Value);
					}
				});
			} catch (Exception ex) {
				Logger.LogVerbose("Error playing audio -- " + ex.Message);
			}
		}

		internal static void PlayAudio(string pAudio, Vector3? pLocation = null)
		{
			try {
				GameFiber.StartNew(() =>
				{
					if (pLocation == null) {
						LSPD_First_Response.Mod.API.Functions.PlayScannerAudio(pAudio);
					} else {
						LSPD_First_Response.Mod.API.Functions.PlayScannerAudioUsingPosition(pAudio, pLocation.Value);
					}
				});
			} catch (Exception ex) {
				Logger.LogVerbose("Error playing audio -- " + ex.Message);
			}
		}

		internal class AudioFile
		{
			internal AudioFile(string pClassName, Enum pAudioFile)
			{
				_class = pClassName;
				_fileEnum = pAudioFile;
			}

			private Enum _fileEnum;
			internal Enum FileEnum {
				get { return _fileEnum; }
				set { _fileEnum = value; }
			}

			private string _class;
			internal string AudioClass {
				get { return _class; }
				set { _class = value; }
			}
		}

	}

}